.ci/magician/github/reviewer_assignment.go

```diff
@@ -66,3 +67,33 @@ func FormatReviewerComment(newPrimaryReviewer string) string {
 	})
 	return sb.String()
 }
+
+var reviewerCommentRegex = regexp.MustCompile("@(?P<reviewer>[^,]*), a repository maintainer, has been assigned to review your changes.")
+
+// FindReviewerComment returns the comment which mentions the current primary reviewer and the reviewer's login,
+// or an empty comment and empty string if no such comment is found.
+// comments should only include comments by the magician in the current PR.
+func FindReviewerComment(comments []PullRequestComment) (PullRequestComment, string) {
+	var newestComment PullRequestComment
+	var currentReviewer string
+	for _, comment := range comments {
+		if !newestComment.CreatedAt.IsZero() && comment.CreatedAt.Before(newestComment.CreatedAt) {
+			// Skip comments older than the newest comment.
+			continue
+		}
+		names := reviewerCommentRegex.SubexpNames()
+		matches := reviewerCommentRegex.FindStringSubmatch(comment.Body)
+		fmt.Println(comment, matches)
+		if len(matches) < len(names) {
+			// Skip comments that don't match regex.
+			continue
+		}
+		for i, name := range names {
+			if name == "reviewer" {
+				newestComment = comment
+				currentReviewer = matches[i]
+			}
+		}
```

Comment:
I think this could be simplified a bit.

```suggestion
		matches := reviewerCommentRegex.FindStringSubmatch(comment.Body)
		fmt.Println(comment, matches)
		if matches == nil {
			// Skip comments that don't match regex.
			continue
		}
		newestComment = comment
		currentReviewer = matches[1]
```

Reply:
I hear you, and I wish golang provided an easier way to access the matches by group name, but this implementation is currently a bit convoluted considering there's only one subgroup in the regex. It would be a bit more clear if you did something like [make a map of all the named groups and then extracted the value from that](https://stackoverflow.com/questions/20750843/using-named-matches-from-go-regex) but that also feels pretty heavy for what's happening here.

Reply:
We could factor the map logic out into a separate util library for handling regex but this particular case doesn't provide a strong justification for it.

Reply:
Just to close this discussion out - I'm not blocking the merge based on this.